snap/info_snap_yaml.go

```diff
@@ -214,6 +215,10 @@ func infoSkeletonFromSnapYaml(y snapYaml) *Info {
 	if len(y.Architectures) != 0 {
 		architectures = y.Architectures
 	}
+	systemUsers := []string{}
+	if len(y.SystemUsers) > 0 {
+		systemUsers = y.SystemUsers
```

Comment:
do we want some validation that the users look like valid system users at all?  we do tend to validate for well formed names around here

Reply:
@pedronis - I can do that but was already doing this in checkSystemUsers() in overlord/snapstate/check_snap.go. The current check is extremely strict: it is either `daemon` or it fails.

Reply:
yes, but these are different layers, I think it's better to also do a more general validation here on would be a valid name, we should also possibly prohibit names starting with snap_ ?

Reply:
> yes, but these are different layers, I think it's better to also do a more general validation here on would be a valid name, we should also possibly prohibit names starting with snap_ ?

I'm not sure that this is the wrong layer is true. AIUI, all infoSkeletonFromSnapYaml() is doing is reading in yaml and it doesn't do any error checking itself (indeed, it doesn't return error or anything) and it is relying on later checks (ie, those in check_snap.go like the one I implemented) to validate the yaml is correct.

To address your concern for 'snap.', etc, I'll update checkSystemUsers to do even more checks.

Reply:
I've adjust check_snap.go to have additional checks. This PR cherrypicked https://github.com/snapcore/snapd/pull/6773 and https://github.com/snapcore/snapd/pull/6774 in order to support that.